astroquery/oac/core.py

```diff
@@ -0,0 +1,602 @@
+        >>> test_radius = 10*u.arcsec
+        >>> test_height = 10*u.arcsec
+        >>> test_width = 10*u.arcsec
+
+        An example cone search:
+        >>> photometry = OAC.query_region(coordinates=test_coords,
+        ...                               radius=test_radius,
+        ...                               quantity="photometry",
+        ...                               attribute=["time", "magnitude",
+        ...                                          "e_magnitude", "band",
+        ...                                          "instrument"])
+
+        An example box search:
+        >>> photometry = OAC.query_region(coordinates=test_coords,
+        ...                               width=test_width, height=test_height,
+        ...                               quantity="photometry",
+        ...                               attribute=["time", "magnitude",
+        ...                                          "e_magnitude", "band",
+        ...                                          "instrument"])
+
+        These searches can be refined using the quantities, attributes, and
+        arguments, as with query_object.
+
+        """
+        # Default object name used for coordinate-based queries
+        event = 'catalog'
+
+        request_payload = self._args_to_payload(event,
+                                                quantity,
+                                                attribute,
+                                                argument,
+                                                data_format)
+
+        # Check that coordinate object is a valid astropy coordinate object
+        # Criteria/Code from ../sdss/core.py
+        if (not isinstance(coordinates, list) and
+            not isinstance(coordinates, Column) and
+            not (isinstance(coordinates, commons.CoordClasses) and
+                 not coordinates.isscalar)):
+
+            request_payload['ra'] = coordinates.ra.deg
+            request_payload['dec'] = coordinates.dec.deg
+
+        else:
+            try:
+                request_payload['ra'] = coordinates[0]
+                request_payload['dec'] = coordinates[1]
+            except Exception:
+                raise ValueError("Please check format of input coordinates.")
+
+        if ((radius is None) and (height is None) and (width is None)):
+            raise ValueError("Please enter a radius or width/height pair.")
+
+        if (radius is not None and (height is not None or width is not None)):
+            raise ValueError("Please specify ONLY a radius or "
+                             "height/width pair.")
+
+        if ((radius is None) and ((height is None) or (width is None))):
+            raise ValueError("Please enter both a width and height "
+                             "for a box search.")
+
+        # Check that any values are in the proper format.
+        # Criteria/Code from ../sdss/core.py
```

Comment:
cc @bsipocz - perhaps we should update `commons.radius_to_unit` to handle this case more generally?  https://github.com/astropy/astroquery/blob/master/astroquery/utils/commons.py#L110

(@pcowpert - you don't need to do anything here, this is a note-to-selves that there is probably a more concise and general way to do what you've done that we should have implemented already...)